File: 백준/Silver/1697. 숨바꼭질/숨바꼭질.py
from collections import deque

def bfs(start, target):
    visited = [False] * 100001
    queue = deque([(start, 0)])
    visited[start] = True
    
    while queue:
        pos, time = queue.popleft()
        
        if pos == target:
            return time
        
        for next_pos in [pos - 1, pos + 1, pos * 2]:
            if 0 <= next_pos <= 100000 and not visited[next_pos]:
                visited[next_pos] = True
                queue.append((next_pos, time + 1))

N, K = map(int, input().split())
print(bfs(N, K))

File: 백준/Silver/1697. 숨바꼭질/숨바꼭질.js
function bfs(start, target) {
    const visited = new Array(100001).fill(false);
    const queue = [[start, 0]];
    visited[start] = true ;
    
    let front = 0;
    while (front < queue.length)  {
        const [pos, time] = queue[front++];
        
        if (pos === target) {
            return time;
        }
        
        for (const nextPos of [pos - 1, pos + 1, pos * 2]) {
            if ((nextPos >= 0 && nextPos <= 100000) && !visited[nextPos]) {
                visited[nextPos] = true;
                queue.push([nextPos, time + 1]);
            }
        }
    }
}

const [N, K] = require('fs').readFileSync(0).toString().trim().split(' ').map(Number);
console.log(bfs(N, K));